.github/workflows/test_minimum_versions.yml

```diff
@@ -0,0 +1,44 @@
+name: Minimum version tests
```

Comment:
This new job will install all dependencies at their minimum supported versions.
If this job breaks, that means we are claiming to support e.g. `qiskit>=1.1` but actually need e.g. `qiskit>=1.4`.